### DISCUSSION TITLE
Configure A2DP Source SBC encoding parameters?

### DISCUSSION THREAD

**Discussion by amondit** [pschatzmann/arduino-audio-tools#722] [Q&A] [answered]
Hello,

Thanks for your work on those libs, it's fantastic!
I'm currently setting up a nodemcu-32s with an I2S ADC (similar 32b one that you have in some of your examples, only newer version based on WM8782) and I am configuring it to stream using A2DP to a custom audio receiver (ubuntu minipc using bluez/pipewire).

I was wondering if we could be able to tweak the ESP32 A2DP Source config to use a higher SBC bitrate (ideally bitpool = 38, 16 blocks in a frame, 8 frequency bands to achieve 452 kb/s bitrate in 44.1KHz). Do you know if the API exposes this possibility (or there might be hw limitations that could prevent this anyway?).

Greetings from Lausanne!

**Comment by pschatzmann** [comment, marked as answer]
No, the ESP32 A2DP does not provide any possibility of this sort. 

Espressif also does not provide the source code, so I don't really see any possibility in Arduino.

Quite recently I have discovered this project https://github.com/bluekitchen/btstack that should work on an ESP32 and provides the full source code. It is not working in Arduino however...
